mtcpclient/calculate.go

```diff
@@ -39,13 +45,19 @@ func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionS
 }
 
 func (gc GroupOfConnections) calculateStdDev(status tcpclient.ConnectionStatus, mr metricsCollectionStats) time.Duration {
+	// TODO: passing the whole mr struct looks overkilling, given we only want a single value, the average, and maybe
```

Comment:
I agree, you only need the average, but I was also using the numberOfConnections that you have replaced by nItems and recalculate again within the loop

Reply:
See previous comment: making functions as autonomous as possible is a good principle from my point of view. As well as making dependencies as explicit as possible...

Reply:
yes, I also agree